tests/instruments/test_all_instruments.py

```diff
@@ -54,14 +54,97 @@
 init = ["ThorlabsPM100USB", "Keithley2700", "TC038", "Agilent34450A",
         "AWG401x_AWG", "AWG401x_AFG", "VARX", "HP8116A"]
 # Instruments which require more input arguments
-init.extend(["Instrument", "ATSBase"])
+name_required = ["ATSBase"]
 
 
 @pytest.mark.parametrize("cls", devices)
-def test_args(cls):
+def test_adapter_arg(cls):
     "Test that every instrument has adapter as their input argument"
     if cls.__name__ in proper_adapters:
         pytest.skip(f"{cls.__name__} does not accept an Adapter instance.")
-    elif cls.__name__ in init:
+    elif cls.__name__ in (*init, *name_required, "Instrument"):
         pytest.skip(f"{cls.__name__} requires communication in init.")
     cls(adapter=MagicMock())
```

Comment:
```suggestion
    elif cls.__name__ in (*init):
        pytest.skip(f"{cls.__name__} requires communication in init.")
    elif cls.__name__ in (*name_required, "Instrument"):
        pytest.skip(f"{cls.__name__} requires more input arguments.")
    cls(adapter=MagicMock())
```
To keep the skip messages accurate.
Why does plain `Instrument` not work, here?

Reply:
Because `Instrument` needs a `name` argument.
If I supply it for all, the other instruments don't like it.

Reply:
I've been following this discussion for a while and wanted to quickly jump in on this point specifically. during #61, I was playing around with this part of the `SPDBase` and had originally included it here, but a comment from @bilderbuchi meant I was able to remove the SPDBase from the equivalent of `name_required` at the time.

> My only excuse is that it must have been too late to notice that comment :-P
> 
> What if you add `kwargs.setdefault('name', 'Siglent SPDxxxxX instrument Base Class')` in `SPDBase`'s init, just before you call `super()__init__`? This should enable using it in tests, but still result in the correct names for the child classes?

Can I just quickly ask to understand why the same thing does not apply here for `ATSBase` or `Instrument` with regards to `name` if a `kwargs.setdefault(...)` has been set for both classes? (I did try to ask this before but I think it went unnoticed)

Reply:
`Instrument` does not have a default name (in order to enforce a name in the instruments, I think).

Atsbase has to be changed, yet.

Reply:
I removed that category and added an individual test for Instrument name (with a proper message).

Another option would be, to allow the instrument to have a default name (which I do not like).

Reply:
Perfect, thanks!